beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -419,269 +410,229 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
         );
     }
 
-    pub fn single_blob_lookup_response(
-        &mut self,
-        id: Id,
+    /// Consolidates error handling for `single_lookup_response`. An `Err` here should always mean
+    /// the lookup is dropped.
+    fn single_lookup_response_inner<R: RequestState<Current, T>>(
+        &self,
         peer_id: PeerId,
-        blob: Option<Arc<BlobSidecar<T::EthSpec>>>,
+        response: Option<R::ResponseType>,
         seen_timestamp: Duration,
-        cx: &mut SyncNetworkContext<T>,
-    ) {
-        let stream_terminator = blob.is_none().into();
-
+        cx: &SyncNetworkContext<T>,
+        mut lookup: SingleBlockLookup<Current, T>,
+    ) -> Result<SingleBlockLookup<Current, T>, LookupRequestError> {
+        let response_type = R::response_type();
         let log = self.log.clone();
+        let expected_block_root = lookup.block_root();
+        let request_state = R::request_state_mut(&mut lookup);
+
+        match request_state.verify_response(expected_block_root, response) {
+            Ok(Some(verified_response)) => {
+                self.handle_verified_response::<Current, R>(
+                    seen_timestamp,
+                    cx,
+                    BlockProcessType::SingleBlock { id: lookup.id },
```

Comment:
Do this need to be generic, or is this always a `SingleBlock` process type here?

Reply:
This is pretty confusing, but it should always be `SingleBlock`. The reason is that it's used in scenarios where we're waiting for the entire block + all blobs to arrive before sending to processing.

Reply:
we could potentially make this less confusing by added a new variant to `BlockProcessType` that is specific to when we have all blocks and blobs